### PR TITLE
APR-1303 - Address check

### PR DESCRIPTION
**Pull request by MakCom**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        private readonly IInternalQnaApiClient _qnaApiClient;
+        private readonly ILogger<RoatpGatewayOrganisationChecksController> _logger;
+
+        public RoatpGatewayOrganisationChecksController(IInternalQnaApiClient qnaApiClient, ILogger<RoatpGatewayOrganisationChecksController> logger)
+        {
+            _qnaApiClient = qnaApiClient;
+            _logger = logger;
+        }
+        
+        [Route("Gateway/Page/QnaCompanyAddress/{applicationId}")]
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs]: Organisation may not be a company, they could be a sole trader or a charity. OrganisationAddress would be a better naming convention.

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            var returnAddress = new GetQnaCompanyAddressResult { Address = applyAddress };
+
+
+            return Ok(returnAddress);
+        }
+    }
+
+    public class GetQnaCompanyAddressResult
+    {
+        public string Address { get; set; }
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs]: Personally, I think this API endpoint should return all the address fields above, not a concatenated address string. The presentation layer in the consumer of the API can then decide how to display it.

**Comment by MichaelYoung1981** [comment]
This cannot be merged without unit tests.

**Comment by MakCom** [comment]
> .
@MichaelYoung1981 
I'm writing the test right now.

**Comment by MichaelYoung1981** [comment]
> > .
> > @MichaelYoung1981
> > I'm writing the test right now.

Thank you, just didn't want the approval to mean the test was missed. 👍

**Pull request closed without merging**